### DISCUSSION TITLE
How getAFeature works and can give an example for override getAFeature

### DISCUSSION THREAD

**Discussion by badnoodle** [github/codeql#10813] [Q&A]
How getAFeature works in Configuration for java?
I don't see any example for custom getAFeature, can I want an example for override getAFeature

**Comment by jketema** [comment]
Thanks for your question. You want to override `getAFeature` and say that result is an instance of one of `FeatureHasSourceCallContext`, `FeatureHasSinkCallContext`, or `FeatureEqualSourceSinkCallContext`.

So for example:
```codeql
override FlowFeature getAFeature() { result instanceof FeatureHasSourceCallContext }
```

**Reply by badnoodle** [reply]
thanks for your replay.
I want to know when I should write a custom getAFeature?

**Reply by jketema** [reply]
You should add a `getAFeature` when you want to restrict flow paths as described in [the documentation](https://codeql.github.com/codeql-standard-libraries/java/semmle/code/java/dataflow/internal/DataFlowImpl.qll/predicate.DataFlowImpl$Configuration$getAFeature.0.html).
